src/imputors/substitute.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    Substitute(; statistic=nothing)
+    Substitute(; robust=true, weights=nothing)
+
+Substitute missing values with a summary statistic over the non-missing values.
+
+# Keyword Arguments
+* `statistic`: A summary statistic function to be applied to the non-missing values.
+  This function should return a value of the same type as the input data `eltype`.
+  If this function isn't passed in then the `defaultstats` function is used to make a
+  best guess.
+* `robust`: Whether to use `median` or `mean` for continuous datasets in `defaultstats`
```

Comment:
This is a bit missleaing.
It also changes categorical data sets.

Reply:
I think I misunderstood.
I find this a bit confusing.
This if that unless the other

Should we just have two types?
One for robust and one for not?

Reply:
or maybe we should just embrace being opinionated and default to `median`?

Reply:
https://github.com/invenia/Impute.jl/issues/81